test/resources/index.ts

```diff
@@ -140,8 +139,9 @@ export const generateTestCase = function (
   return new TestCase(
     testCaseInput.fundingAMMToken0,
     testCaseInput.fundingAMMToken1,
-    _.cloneDeep(testCaseInput.sellOrdersToken0), // <-- deep copy needed as solveTestCase can modify the orders
-    _.cloneDeep(testCaseInput.sellOrdersToken1),
+    // deep copy needed as solveTestCase can modify the orders
+    testCaseInput.sellOrdersToken0.map((order) => order.clone()),
+    testCaseInput.sellOrdersToken1.map((order) => order.clone()),
```

Comment:
The assumption here is that `solveTestCase` only changes the orders up to the depth of the cloning done here?

Reply:
:+1:

Reply:
Would be nice if it was not mutating the input.

Reply:
Also agree, I will look into it in a follow up PR.